### PR TITLE
Route properties

### PR DESCRIPTION
**Pull request by ssahani**
route add MPLS TTL propagate

### CODE REVIEW

File: src/libsystemd/sd-netlink/netlink-types.c [hunk 1]
```diff
@@ -622,23 +622,23 @@ static const NLType rtnl_route_types[] = {
...
-        RTA_TABLE,
-        RTA_MARK,
-        RTA_MFC_STATS,
-        RTA_VIA,
-        RTA_NEWDST,
-*/
+        [RTA_CACHEINFO]         = { .type = NETLINK_TYPE_U32 },
+        [RTA_TABLE]             = { .type = NETLINK_TYPE_U32 },
+        [RTA_MARK]              = { .type = NETLINK_TYPE_U32 },
+        [RTA_MFC_STATS]         = { .type = NETLINK_TYPE_U32 },
```
Comment (yuwata) [inline src/libsystemd/sd-netlink/netlink-types.c]: U64?

File: src/libsystemd/sd-netlink/netlink-types.c [hunk 2]
```diff
@@ -622,23 +622,23 @@ static const NLType rtnl_route_types[] = {
...
         [RTA_FLOW]              = { .type = NETLINK_TYPE_U32 }, /* 6? */
-/*
-        RTA_CACHEINFO,
-        RTA_TABLE,
-        RTA_MARK,
-        RTA_MFC_STATS,
-        RTA_VIA,
-        RTA_NEWDST,
-*/
+        [RTA_CACHEINFO]         = { .type = NETLINK_TYPE_U32 },
```
Comment (yuwata) [inline src/libsystemd/sd-netlink/netlink-types.c]: .size = sizeof(struct rta_cacheinfo)

File: src/network/networkd-route.c [hunk 3]
```diff
@@ -1300,6 +1307,44 @@ int config_parse_route_mtu(
...
+        assert(rvalue);
+        assert(data);
+
+        r = route_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        k = parse_boolean(rvalue);
+        if (k < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, k, "Failed to TTLPropagate= value, ignoring: %s", rvalue);
```
Comment (yuwata) [inline src/network/networkd-route.c]: Failed to _parse_

**Comment by ssahani** [comment]
updated. Thanks for the review @yuwata

File: src/libsystemd/sd-netlink/netlink-types.c [hunk 4]
```diff
@@ -622,23 +622,23 @@ static const NLType rtnl_route_types[] = {
...
         [RTA_FLOW]              = { .type = NETLINK_TYPE_U32 }, /* 6? */
-/*
-        RTA_CACHEINFO,
-        RTA_TABLE,
-        RTA_MARK,
-        RTA_MFC_STATS,
-        RTA_VIA,
-        RTA_NEWDST,
-*/
+        [RTA_CACHEINFO]         = { .type = sizeof(struct rta_cacheinfo) },
```
Comment (yuwata) [inline src/libsystemd/sd-netlink/netlink-types.c]: `.type` -> `.size`

Comment (ssahani) [reply src/libsystemd/sd-netlink/netlink-types.c]: yes missed.

**Comment by ssahani** [comment]
Updated PTAL @yuwata

**Pull request merged**